Sources/Nats/NatsSubscription.swift

```diff
@@ -150,4 +160,26 @@ public class NatsSubscription: AsyncSequence {
         }
         return try await self.conn.unsubscribe(sub: self, max: after)
     }
+
+    // validateSubject will do a basic subject validation.
+    // Spaces are not allowed and all tokens should be > 0 in length.
+    private static func validSubject(_ subj: String) -> Bool {
+        let whitespaceCharacterSet = CharacterSet.whitespacesAndNewlines
+        if subj.rangeOfCharacter(from: whitespaceCharacterSet) != nil {
+            return false
+        }
+        let tokens = subj.split(separator: ".")
+        for token in tokens {
+            if token.isEmpty {
```

Comment:
should we add wildcard checks? e.g. these tokens `abc*`, `>abc` and `a*bc` are invalid I think, no?

Reply:
I don't agree actually. It is legal to send wildcards within a subject name, even if it's not recommended. I checked with CLI, and you can normally publish ad subscribe to the subjects you mentioned above.

Additionally, we should not do extensive validation of subjects in the client on hotpaths (I treat core NATS pub/sub as such), as long as those subjects don't break server protocol (e.g. `\n` needs to be checked).

Reply:
> It is legal to send wildcards within a subject name, even if it's not recommended. I checked with CLI, and you can normally publish ad subscribe to the subjects you mentioned above.

makes sense. thanks.